R/Attributes.R

```diff
@@ -588,8 +588,52 @@ compileAttributes <- function(pkgdir = ".", verbose = getOption("verbose")) {
 
 ## built-in OpenMP plugin
 .plugins[["openmp"]] <- function() {
-    list(env = list(PKG_CXXFLAGS="-fopenmp",
-                    PKG_LIBS="-fopenmp"))
+
+    # Default flags for non-macOS systems
+    cxxflags <- "-fopenmp"
+    libs <- "-fopenmp"
+
+    # Check if we're on macOS
+    if (Sys.info()[["sysname"]] == "Darwin") {
+        # Get the C++ compiler from R CMD config CXX
+        r_cmd <- file.path(R.home("bin"), "R")
+        cxx <- tryCatch({
+            system2(r_cmd, c("CMD", "config", "CXX"), stdout = TRUE, stderr = FALSE)
+        }, error = function(e) "")
+
+        if (length(cxx) > 0 && nzchar(cxx[1])) {
+            # Check compiler version using full command (this includes any flags)
+            compiler_version <- tryCatch({
+                system(paste(cxx[1], "--version"), intern = TRUE, ignore.stderr = TRUE)
+            }, error = function(e) character(0))
+
+            if (length(compiler_version) > 0) {
+                # Check for Apple clang
+                is_apple_clang <- any(grepl("Apple", compiler_version, ignore.case = TRUE))
```

Comment:
This check also succeeds (and is checked in the conditional) for Homebrew's build of clang, since it has

```
> writeLines(compiler_version)
Homebrew clang version 21.1.6
Target: arm64-apple-darwin25.1.0
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/llvm/21.1.6/bin
Configuration file: /opt/homebrew/etc/clang/arm64-apple-darwin25.cfg
```

and so `apple` is picked up in the grep invocation from the target triple.